src/sparql/qc/mondo/qc-multiple-gene-associations.sparql

```diff
@@ -0,0 +1,34 @@
+PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+PREFIX owl: <http://www.w3.org/2002/07/owl#>
+PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+PREFIX obo: <http://purl.obolibrary.org/obo/>
+PREFIX oboInOwl: <http://www.geneontology.org/formats/oboInOwl#>
+
+# Get classes that have more than 1 gene association (either subClassOf or equivalentClass) with RO:0004003 property
+
+SELECT DISTINCT ?entity ?label (GROUP_CONCAT(DISTINCT ?geneIdentifier; separator=", ") AS ?geneIdentifiers)
+WHERE {
+  {
+    # subClassOf association
+    ?entity rdfs:subClassOf ?restriction ;
+            rdfs:label ?label .
+    
+    ?restriction rdf:type owl:Restriction ;
+                 owl:onProperty obo:RO_0004003 ;
+                 owl:someValuesFrom ?geneIdentifier .
+  }
+  UNION
+  {
+    # equivalentClass association
+    ?entity owl:equivalentClass ?equivClass ;
+            rdfs:label ?label .
+    
+    ?equivClass owl:intersectionOf/rdf:rest*/rdf:first ?component .
+    
+    ?component rdf:type owl:Restriction ;
+               owl:onProperty obo:RO_0004003 ;
+               owl:someValuesFrom ?geneIdentifier .
+  }
+}
```

Comment:
My feeling is that in a UNION query, the bindings are renewed, so ?entity in the first union does not have to be bound to the same ?entity in the second union query. To mitigate this, you can move the ?label outside the UNION, like so:

```
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX owl: <http://www.w3.org/2002/07/owl#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX obo: <http://purl.obolibrary.org/obo/>
PREFIX oboInOwl: <http://www.geneontology.org/formats/oboInOwl#>

SELECT DISTINCT ?entity ?label (GROUP_CONCAT(DISTINCT ?geneIdentifier; separator=", ") AS ?geneIdentifiers)
WHERE {
  {
    # subClassOf association
    ?entity rdfs:subClassOf ?restriction .
    ?restriction rdf:type owl:Restriction ;
                 owl:onProperty obo:RO_0004003 ;
                 owl:someValuesFrom ?geneIdentifier .
  }
  UNION
  {
    # equivalentClass association
    ?entity owl:equivalentClass ?equivClass .
    ?equivClass (owl:intersectionOf/rdf:rest*/rdf:first) ?component .
    ?component rdf:type owl:Restriction ;
               owl:onProperty obo:RO_0004003 ;
               owl:someValuesFrom ?geneIdentifier .
  }

  # Ensure consistent labeling, if available
  ?entity rdfs:label ?label .
}
GROUP BY ?entity ?label
HAVING (COUNT(DISTINCT ?geneIdentifier) > 1)
```

Now the entity is bound outside the union, and inside they will bind to the same.

Could be wrong though, just tickling the back of my head?

Reply:
I had these as separate query files before, one qc query to check if there is more than one gene association stated using the equivalentTo restriction and one qc query to check if there is more than one gene association stated as a subClassOf relationship, so I'm not trying to check if the same entity is used in both. After having each query work separately, I though you might ask me to combine them so I did... but they can also easily be separate qc queries.

Reply:
I'm not sure if we are both trying to get the query to check the same thing because I don't see why the ?entity needs to be the same for both the equivalentTo restriction and subClassOf. I want the qc check to fail if either of these ways of representing a gene association in Mondo have >1 gene association. So fail if >1 gene association represented as a subClassOf or fail if >1 gene association represented in the logical definition.

Given that, do you think the query still needs to be changed?

Reply:
I think you might be right.. I retract my comment, lgtm!

Reply:
well someone needs to officially approve the PR :)